Framework/DataHandling/src/LoadSwans.cpp

```diff
@@ -0,0 +1,298 @@
+
+  g_log.information() << "Moving detector " << componentName << " " << distance
+                      << " meters and " << angle << " degrees." << std::endl;
+
+  LoadHelper helper;
+  constexpr double deg2rad = M_PI / 180.0;
+  V3D pos = helper.getComponentPosition(m_ws, componentName);
+  double angle_rad = angle * deg2rad;
+  V3D newpos(distance * sin(angle_rad), pos.Y(), distance * cos(angle_rad));
+  helper.moveComponent(m_ws, componentName, newpos);
+
+  // Apply a local rotation to stay perpendicular to the beam
+  const V3D axis(0.0, 1.0, 0.0);
+  Quat rotation(angle, axis);
+  helper.rotateComponent(m_ws, componentName, rotation);
+}
+
+/**
+ * Load the data into a map. The map is indexed by pixel id (0 to 128*128-1 =
+ * m_detector_size)
+ * The map values are the events TOF
+ * @returns the map of events indexed by pixel index
+ */
+std::map<uint32_t, std::vector<uint32_t>> LoadSwans::loadData() {
+
+  std::string filename = getPropertyValue("FilenameData");
+  std::ifstream input(filename, std::ifstream::binary | std::ios::ate);
+  input.seekg(0);
+
+  m_ws->initialize(m_detector_size, 1, 1);
+
+  std::map<uint32_t, std::vector<uint32_t>> eventMap;
+
+  while (input.is_open()) {
+    if (input.eof())
+      break;
+    uint32_t tof = 0;
+    // input.read((char *)&tof, sizeof(tof));
+    input.read(reinterpret_cast<char *>(&tof), sizeof(tof));
+    tof -= static_cast<uint32_t>(1e9);
+    tof = static_cast<uint32_t>(tof * 0.1);
+
+    uint32_t pos = 0;
+    // input.read((char *)&pos, sizeof(pos));
+    input.read(reinterpret_cast<char *>(&pos), sizeof(pos));
+    if (pos < 400000) {
+      g_log.warning() << "Detector index invalid: " << pos << std::endl;
+      continue;
+    }
+    pos -= 400000;
+    eventMap[pos].push_back(tof);
+  }
+
+  return eventMap;
+}
+
+/**
+ * Load metadata file witch to date is just a line of of double values
+ * Parses this file and put it into a vector
+ * @return vector with the file contents
+ */
+std::vector<double> LoadSwans::loadMetaData() {
+
+  std::vector<double> metadata;
+  std::string filename = getPropertyValue("FilenameMetaData");
+
+  std::ifstream infile(filename);
+
+  if (infile.fail()) {
+    g_log.error("Error reading file " + filename);
+    throw Exception::FileError("Unable to read data in File:", filename);
+  }
+
+  std::string line;
+  while (getline(infile, line)) {
+    // line with data, need to be parsed by white spaces
+    if (!line.empty() && line[0] != '#') {
+      g_log.debug() << "Metadata parsed line: " << line << std::endl;
+      std::vector<std::string> dataLine;
+      boost::trim_if(
+          line, boost::is_any_of("\t ")); // could also use plain boost::trim
+      boost::split(dataLine, line, boost::is_any_of("\t "),
+                   boost::token_compress_on);
+      for (auto beg = dataLine.begin(); beg != dataLine.end(); ++beg) {
+        std::stringstream ss;
+        ss << *beg;
+        double d;
+        ss >> d;
+        metadata.push_back(d);
```

Comment:
This can be done on one line with `boost::lexical_cast`
http://www.boost.org/doc/libs/1_60_0/doc/html/boost_lexical_cast/examples.html#boost_lexical_cast.examples.strings_to_numbers_conversion